### PR TITLE
Add consumeOrFail().

### PR DESCRIPTION
**Pull request by dereuromark**
Took the liberty of finishing up https://github.com/cakephp/cakephp/issues/10685

### CODE REVIEW

**Comment by inoas** [comment]
Would it make sense to replace the few places where consume() is being used within the core framework with consumeOrFail()??

**Comment by dereuromark** [comment]
This can be a follow up PR that runs the tests then separately to make sure nothing goes wrong.

**Comment by dereuromark** [comment]
Yeah, I fixed that in the 2nd commit.

File: tests/TestCase/Core/ConfigureTest.php [hunk 1]
```diff
@@ -592,4 +592,29 @@ public function testConsumeEmpty()
...
+     *
+     * @return void
+     */
+    public function testConsumeOrFail()
+    {
+        $expected = 'ok';
+        Configure::write('This.Key.Exists', $expected);
+        $result = Configure::consumeOrFail('This.Key.Exists');
+        $this->assertEquals($expected, $result);
+    }
```
Comment (ADmad) [inline tests/TestCase/Core/ConfigureTest.php]: An assertion testing that the value has been consumed and cannot be read again would be nice.

Comment (dereuromark) [reply tests/TestCase/Core/ConfigureTest.php]: I think that part is covered in consume tests itself already.

**Comment by dereuromark** [comment]
Docs @ https://github.com/cakephp/docs/pull/5641

**Pull request merged**